packages/sync-actions/src/product-actions.js

```diff
@@ -256,6 +220,30 @@ export function actionsMapPrices (diff, oldObj, newObj) {
     .concat(addPriceActions)
 }
 
+export function generateChangeMasterVariantAction (oldObj, newObj) {
+  const createChangeMasterVariantAction = variantId => ({
+    action: 'changeMasterVariant',
+    variantId,
+  })
+  const extractMasterVariantId = (fromObj) => {
+    const variants = Array.isArray(fromObj.variants) ? fromObj.variants : []
+
+    return variants[0] ? variants[0].id : undefined
```

Comment:
This feels a bit wonky but eases the code further down. Generally caused by variants not having to be set and hence the `id` not necessarily having to be present.